src/utils/normalizeModuleFederationOptions.ts

```diff
@@ -232,36 +224,31 @@ export type ModuleFederationOptions = {
   library?: any;
   name: string;
   // remoteType?: string;
-  remotes?:
-  | Record<
-    string,
-    | string
-    | RemoteObjectConfig
-  >
-  | undefined;
+  remotes?: Record<string, string | RemoteObjectConfig> | undefined;
   runtime?: any;
   shareScope?: string;
   shared?:
-  | string[]
-  | Record<
-    string,
-    | string
-    | {
-      name?: string;
-      version?: string;
-      shareScope?: string;
-      singleton?: boolean;
-      requiredVersion?: string;
-      strictVersion?: boolean;
-    }
-  >
-  | undefined;
+    | string[]
+    | Record<
+        string,
+        | string
+        | {
+            name?: string;
+            version?: string;
+            shareScope?: string;
+            singleton?: boolean;
+            requiredVersion?: string;
+            strictVersion?: boolean;
+          }
+      >
+    | undefined;
   runtimePlugins?: string[];
   getPublicPath?: any;
   implementation?: any;
   manifest?: ManifestOptions | boolean;
   dev?: any;
   dts?: any;
+  shareStrategy: ShareStrategy;
```

Comment:
Should this be optional?

Reply:
Yup, nice catch. Would you like to drop a PR for that? you deserve it

Reply:
Certainly! https://github.com/module-federation/vite/pull/130